### PR TITLE
Hw20 mockito test

### PR DESCRIPTION
**Pull request by vluchyk**

### CODE REVIEW

**Review by Danylo-Honcharenko — APPROVED** [review]

**Review by rudendan — APPROVED** [review]

File: src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        expected = AddressDto.builder().id(1).number("1000").street("Drayton str").apartmentNumber(7).build();
+        updated = AddressDto.builder().id(1).number("2000").street("Seven Hills str").apartmentNumber(1).build();
+    }
+
+    @Test
+    public void getTest() throws ObjectNotFoundException {
+        Mockito.when(addressRepository.findById(any())).thenReturn(Optional.of(AddressConverter.toEntity(expected)));
+
+        AddressDto result = addressService.get(anyInt());
+        Assertions.assertEquals(expected, result);
```
Comment (mkharkhuta) [inline src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java:45]: In this test you are comparing the expected with the expected from mock object which is not quite correct from a testing point of view. It' better to return just created object in mock (would be more clear in more complex tests)

File: src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    @BeforeEach
+    public void init() {
+        addressService = new AddressService(addressRepository);
+        expected = AddressDto.builder().id(1).number("1000").street("Drayton str").apartmentNumber(7).build();
+        updated = AddressDto.builder().id(1).number("2000").street("Seven Hills str").apartmentNumber(1).build();
+    }
+
+    @Test
+    public void getTest() throws ObjectNotFoundException {
```
Comment (mkharkhuta) [inline src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java:41]: please follow given_when_then naming convention. Given state may be omitted for some cases:
getTest() -> whenGet_thenReturnAddress()
also negative cases:
givenNoAddress_whenGet_thenThrowException()

File: src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        AddressDto result = addressService.save(expected);
+        Assertions.assertEquals(expected, result);
+    }
+
+    @Test
+    public void updateTest() throws ObjectNotFoundException {
+        AddressDto existing = AddressDto.builder()
+                .id(updated.getId())
+                .build();
+        AddressEntity addressEntity = AddressConverter.toEntity(existing);
```
Comment (mkharkhuta) [inline src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java:69]: it is better to directly create an entity without dot

File: src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        Mockito.when(addressRepository.findById(any())).thenReturn(Optional.of(addressEntity));
+        Mockito.when(addressRepository.save(any())).thenReturn(addressEntity);
+
+        AddressDto result = addressService.update(updated);
+        Assertions.assertEquals(updated, result);
```
Comment (mkharkhuta) [inline src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java:74]: more useful to create objects in place, it is a little bit complicated to search for the global object

File: src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    @Test
+    public void updateTest() throws ObjectNotFoundException {
+        AddressDto existing = AddressDto.builder()
+                .id(updated.getId())
+                .build();
+        AddressEntity addressEntity = AddressConverter.toEntity(existing);
+        Mockito.when(addressRepository.findById(any())).thenReturn(Optional.of(addressEntity));
+        Mockito.when(addressRepository.save(any())).thenReturn(addressEntity);
```
Comment (mkharkhuta) [inline src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java:71]: so, the save(...) returns the same as findById(...). is it expected?

**Review by mkharkhuta — COMMENTED** [review]
Good job 👍  heck the comments please. In general:
1. follow given_when_then
2. create objects for tests in place

File: src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java (hunk 5 above)
Comment (vluchyk) [reply src/test/java/com/gmail/luchyk/viktoriia/glovodb/service/AddressServiceTest.java:71]: @mkharkhuta ,
not the same.
image
image

**Pull request merged**